mep3_simulation/webots_data/controllers/opponent_robot/opponent_robot.py

```diff
@@ -13,71 +12,73 @@
 name_field = opponent_node.getField('name')
 opponent_filed=opponent_node.getField('translation')
 name = name_field.getSFString()
-pos=[0,0,0]
 
-positions=[(-0.97, 0.40, 0.175), (-1.5, 0.5, 0.175),
-           (-0.97, 0.64, 0.175), (-1.23, 0.49, 0.175), 
-           (-0.62, 0.49, -0.175),(-0.62, 0.57, 0.175),
-           (-0.62, 0.43, 0.175), (-1.29, 0.46, 0.175)]
+position=[0,0,0]
+
+positions=[(-0.97, 0.40, 10), (-1.5, 0.5, 10),
+           (-0.97, 0.64, 15), (-1.23, 0.49, 20), 
+           (-0.62, 0.49, 10),(-0.62, 0.57, 15),
+           (-0.62, 0.43, 5), (-1.29, 0.46, 10)]
            
-time=[10,10,15,20,10, 5,10,25]
+
 def destination_achieved(curr, dest, epsilon):
-    if (abs(curr[0]-dest[0])<epsilon and abs(curr[1]-dest[1])<epsilon):
-        return True
-    else:
-        return False
+    return (abs(curr[0]-dest[0])<epsilon and abs(curr[1]-dest[1])<epsilon)
+      
  
-def time_achieved(time,pos,dest):
-   ret=False 
-   t=datetime.now().second
-   if(abs(pos[0]-dest[0])<0.05 and abs(pos[1]-dest[1])<0.05):
-     while(t<time):
-         t=datetime.now().second
-     if i>time:
-        ret=True     
-     return ret
+def time_achieved(time_period):
+   ret_value=False
+   t=supervisor.getTime()
+   while(supervisor.getTime()-t<time_period):
```

Comment:
treba zameniti while sa if petljom

Reply:
@MarijaGolubovic mozes li ovo opet pogledati?

U sustini, ako dobro ispratimo, ovo ce vrteti u While stalno i nikada nece vratiti False, nego ce biti u while tokom time_period i jednom vratiti True?

Radi i ovako, ali ne onako kako je zamisljeno, cini mi se da je ovako veoma konfuzno, treba opet analizirati.

Reply:
@miloshunter  kada malo bolje pogledam mislim da je False unutar while nepotreban. Inicijalna vrijednost je False i ostaje False sve dok ne udje u else dio, tj. dok ne prodje zadati vremenski interval.

Reply:
Ako je funkcija zamišljena da čeka neko vreme, pa vrati True kad vreme prođe, onda je ispravnost zadovoljena, ali nisam siguran jel to najbolje rešenje za Webots...


Ako je zamišljena da vraća False dok ne istekne vreme, pa onda vrati True kad istekne, onda nije OK.

Jer dok ne istekne vreme, neće nikad izaći iz petlje i neće vraćati False. 
Kod ovaj bi bio ekvivalentan kodu da samo ispod While petlje bude **return True**, čini mi se



Šta je od ova 2 ideja?

Reply:
Funkcija je zamišljena da čeka dok ne prođe zadati vremenski interval, nakon toga se mijenja  vrijednost flag-a. Ovim bi omogućili prelazak na narednu poziciju.

Reply:
Što se tiče Webots-a, nova verzija mi previše opterećuje procesor. Većinu vremena procesor je na 100% ili oko 100% iskorištenosti. Sa prethodnom verzijom nisam imala ovaj problem, pokretala sam sam isti kod.

Reply:
U tom slucaju je onda OK. Samo bih preporucio da se zove drugacije mozda sama funkcija, ali nije kriticno.

Mislim da je sada sve OK.